pkg/designer/domain_test.go

```diff
@@ -0,0 +1,198 @@
+							Type: "smbios",
+							BIOS: []apiv1.Entry{
+								{Name: "vendor", Value: "ACME"},
+							},
+							System: []apiv1.Entry{
+								{Name: "manufacturer", Value: "ACME"},
+								{Name: "product", Value: "RoadRunner"},
+							},
+							BaseBoard: []apiv1.Entry{
+								{Name: "manufacturer", Value: "ACME"},
+								{Name: "product", Value: "Wile E Coyote"},
+							},
+						},
+						Devices: apiv1.Devices{
+							Disks: []apiv1.Disk{
+								{
+									Device: "disk",
+									Source: apiv1.DiskSource{
+										PersistentVolumeClaim: &apiv1.DiskSourcePersistentVolumeClaim{
+											ClaimName: testutil.TestPersistentVolumeClaimISCSI.ObjectMeta.Name,
+										},
+									},
+									Target: apiv1.DiskTarget{
+										Device: "vda",
+										Bus:    "virtio",
+									},
+								},
+								{
+									Device: "disk",
+									Source: apiv1.DiskSource{
+										ISCSI: &apiv1.DiskSourceISCSI{
+											TargetPortal: "127.0.0.1:6543",
+											Lun:          2,
+											IQN:          "iqn.2009-02.com.test:for.all",
+										},
+									},
+									Target: apiv1.DiskTarget{
+										Device: "vdb",
+										Bus:    "virtio",
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			expectXML := strings.Join([]string{
+				`<domain type="kvm">`,
+				`  <name>testvm</name>`,
+				`  <uuid>9838eeb0-663e-485d-a544-3163fb1f0b28</uuid>`,
+				`  <memory unit="MiB">8</memory>`,
+				`  <sysinfo type="smbios">`,
+				`    <system>`,
+				`      <entry name="manufacturer">ACME</entry>`,
+				`      <entry name="product">RoadRunner</entry>`,
+				`    </system>`,
+				`    <bios>`,
+				`      <entry name="vendor">ACME</entry>`,
+				`    </bios>`,
+				`    <baseBoard>`,
+				`      <entry name="manufacturer">ACME</entry>`,
+				`      <entry name="product">Wile E Coyote</entry>`,
+				`    </baseBoard>`,
+				`  </sysinfo>`,
+				`  <os>`,
+				`    <type arch="x86_64" machine="q35">hvm</type>`,
+				`    <bootmenu enabled="yes" timeout="10"></bootmenu>`,
+				`    <smbios mode="sysinfo"></smbios>`,
+				`  </os>`,
+				`  <devices>`,
+				`    <disk type="network" device="disk">`,
+				`      <source protocol="iscsi" name="iqn.2009-02.com.test:for.all/1">`,
+				`        <host transport="tcp" name="127.0.0.1" port="6543"></host>`,
+				`      </source>`,
+				`      <target dev="vda" bus="virtio"></target>`,
+				`    </disk>`,
+				`    <disk type="network" device="disk">`,
+				`      <source protocol="iscsi" name="iqn.2009-02.com.test:for.all/2">`,
+				`        <host transport="tcp" name="127.0.0.1" port="6543"></host>`,
+				`      </source>`,
+				`      <target dev="vdb" bus="virtio"></target>`,
+				`    </disk>`,
+				`  </devices>`,
+				`</domain>`,
+			}, "\n")
+
+			restClient, err := testutil.NewKubeRESTClient(server.URL())
+			Expect(err).NotTo(HaveOccurred())
+
+			domdesign := designer.NewDomainDesigner(restClient, kubeapi.NamespaceDefault)
```

Comment:
Should the DomainDesigner object inherit the namespace from the vm during applySpec rather than being explicitly passed in here?